File: C++/Numbers/Harshad-Number.cpp
#include <bits/stdc++.h>
using namespace std;

bool isharshadNumber(int n){
    int num = n;
    int sum = 0;
    while(n > 0){
        int lastdigit = n % 10;
        sum = sum + lastdigit;
        n = n /10;
    }

    return (num % sum == 0);
}

int main() {
    int n;
    cin>>n;

    bool ans = isharshadNumber(n);

    if(ans){
        cout<<"YES"<<endl;
    }
    else{
        cout<<"No"<<endl;
    }

    return 0;
}

File: Java/Numbers/HarshadNumber.java
public class HarshadNumber {
    public static void main(String[] args) {
        int n = 378;
        System.out.println(harshad(n));
    }

    public static boolean harshad(int num) {
        int sum = 0;
        int temp = num;
        while (num > 0) {
            int lastDigit = num % 10;
            sum += lastDigit;
            num = num / 10;
        }

        if (temp % sum == 0) {
            return true;
        }

        return false;
    }
}
